pkg/proctree/fileinfo.go

```diff
@@ -21,243 +22,379 @@ type FileInfoFeed struct {
 // File Info
 //
 
+const (
+	// string members
+	fileInfoPath changelog.MemberKind = iota
+)
+
+const (
+	// int members
+	fileInfoDev changelog.MemberKind = iota
+	fileInfoCtime
+	fileInfoInode
+	fileInfoInodeMode
+)
+
+var (
+	// fileInfoMutableStringsFlags is a slice with metadata about the mutable string members of a FileInfo.
+	fileInfoMutableStringsFlags = []changelog.MaxEntries{
+		fileInfoPath: 3, // file path
+	}
+
+	// fileInfoMutableIntsFlags is a slice with metadata about the mutable int members of a FileInfo.
+	fileInfoMutableIntsFlags = []changelog.MaxEntries{
+		fileInfoDev:       3, // device number of the file
+		fileInfoCtime:     3, // creation time of the file
+		fileInfoInode:     3, // inode number of the file
+		fileInfoInodeMode: 3, // inode mode of the file
+	}
+)
+
 // FileInfo represents a file.
 type FileInfo struct {
-	path      *ch.Changelog[string] // file path
-	dev       *ch.Changelog[int]    // device number of the file
-	ctime     *ch.Changelog[int]    // creation time of the file
-	inode     *ch.Changelog[int]    // inode number of the file
-	inodeMode *ch.Changelog[int]    // inode mode of the file
-	mutex     *sync.RWMutex
+	mutableStrings *changelog.Entries[string]
+	mutableInts    *changelog.Entries[int]
+	mutex          *sync.RWMutex
 }
```

Comment:
The whole fields of FileInfo are tightly coupled - they all change together (unlike taskInfo) - so what do you think about putting the whole FileInfo struct under one changelog.Entries[FileInfo]? This may reduce memory a bit more plus keeping the original field names of FileInfo for direct access (so we can remove specific setters and getters for each one of them)

Reply:
Just mentioning that in this case flat slice/linked list will almost certainly be better since a slice of slices will be a redundancy (there is only one memberkind). Maybe there should a be a version of entries for a single memberkind usecase.

Reply:
I implemented it in #4427.

Reply:
Got it back here.